cmake/FindWayland.cmake

```diff
@@ -0,0 +1,17 @@
+find_package(PkgConfig)
+
+if(PKG_CONFIG_FOUND)
+    pkg_search_module(WAYLAND_CLIENT IMPORTED_TARGET wayland-client)
+endif()
```

Comment:
Last suggestion: as I just mentioned in https://github.com/yuzu-emu/yuzu/pull/9395#discussion_r1047456172, you could avoid explicitly checking for `PKG_CONFIG_FOUND`; it is done by `pkg_search_module()` automatically. Sorry for being so pedantic 😅️